src/WorldStorage/TreeNBT.h

```diff
@@ -0,0 +1,780 @@
+
+#pragma once
+
+#include "NBTDef.h"
```

Comment:
It would be beneficial to add a comment here explaining how to use the new classes, what is the designed API workflow etc. Since the file declares quite a few classes, one would get lost trying to figure out how to actually use it.